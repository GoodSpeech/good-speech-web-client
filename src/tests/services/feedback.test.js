import {
  compute,
  getScore
} from '../../services/feedback';

describe('Feedback', () => {
  describe('compute', () => {
    it('should bring some results', () => {
      expect(compute('foo', 'bar')).toMatchSnapshot();
      expect(compute(
        'A similar approach can be taken when it comes to testing your React components.',
        'A similar dsf mar be taken when it dome to javi your React components. Instead')
      ).toMatchSnapshot();
    });
  });

  describe('getScore', () => {
    it('bring some score', () => {
      expect(getScore(compute('foo', 'foo'))).toBe(100);
      expect(getScore(compute('foo', 'bar'))).toBe(0);
      expect(getScore(compute(
        'A similar approach can be taken when it comes to, testing your React components. Something',
        'A similar dsf mar be taken when it dome to javi, your React components. Instead'))
      ).toBe(70);
    });
  });
});