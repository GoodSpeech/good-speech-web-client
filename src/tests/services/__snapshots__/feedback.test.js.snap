// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback compute should bring some results 1`] = `
Array [
  Object {
    "position": 0,
    "similarity": 0,
    "value": "foo",
  },
]
`;

exports[`Feedback compute should bring some results 2`] = `
Array [
  Object {
    "position": 0,
    "similarity": 1,
    "value": "A similar ",
  },
  Object {
    "position": 1,
    "similarity": 0,
    "value": "approach",
  },
  Object {
    "position": 3,
    "similarity": 1,
    "value": " ",
  },
  Object {
    "position": 4,
    "similarity": 0,
    "value": "can",
  },
  Object {
    "position": 6,
    "similarity": 1,
    "value": " be taken when it ",
  },
  Object {
    "position": 7,
    "similarity": 0,
    "value": "comes",
  },
  Object {
    "position": 9,
    "similarity": 1,
    "value": " to ",
  },
  Object {
    "position": 10,
    "similarity": 0.25,
    "value": "testing",
  },
  Object {
    "position": 12,
    "similarity": 1,
    "value": " your React components.",
  },
]
`;
